src/routes/dictionaries/+page.svelte

```diff
@@ -46,65 +41,98 @@
     try {
       const uploadedDictionaryOrAdaptation = await effects.uploadDictionaryOrAdaptation(files, data.user);
 
-      if (uploadedDictionaryOrAdaptation !== null) {
-        let seenSet: Set<number> = new Set<number>();
-
-        switch (uploadedDictionaryOrAdaptation.type) {
-          case DictionaryTypes.command_dictionary: {
-            commandDictionaries.updateValue((commandDictionaries: CommandDictionary[]) =>
-              [uploadedDictionaryOrAdaptation.uploadedObject as CommandDictionary, ...commandDictionaries].filter(
-                val => {
-                  if (!seenSet.has(val.id)) {
-                    seenSet.add(val.id);
-                    return true;
-                  }
-
-                  return false;
-                },
-              ),
-            );
-
-            showSuccessToast('Command Dictionary Created Successfully');
-            break;
-          }
-          case DictionaryTypes.param_def:
-            parameterDictionaries.updateValue((parameterDictionaries: ParameterDictionary[]) =>
-              [uploadedDictionaryOrAdaptation.uploadedObject, ...parameterDictionaries].filter(val => {
-                if (!seenSet.has(val.id)) {
-                  seenSet.add(val.id);
-                  return true;
-                }
+      if (uploadedDictionaryOrAdaptation === null) {
+        throw Error('Failed to upload file');
+      }
 
+      let seenSet: Set<number> = new Set<number>();
+      switch (uploadedDictionaryOrAdaptation.type) {
+        case 'COMMAND': {
```

Comment:
Same as the backend, these should be in an `enum`.

Reply:
No problem, I added that